### PR TITLE
BUG: Fix mismatched columns

### LINKED ISSUE #93: BUG: Mismatched fcsv column
## The problem

Moving from #91 to separate issue. Validator throws a server error when the following fcsv is uploaded: [2009bMNI_T1_TM_01.txt](https://github.com/afids/afids-validator/files/4772594/2009bMNI_T1_TM_01.txt).

There appears to be a mismatch of columns / missing information for first two fiducials (AC. PC). See below for the two problematic lines.

```
vtkMRMLMarkupsFiducialNode_0,0.023,-8.165,-16.415,0.000,0.000,0.000,1.000,1,1,1,AC,,vtkMRMLScalarVolumeNode1
vtkMRMLMarkupsFiducialNode_1,0.536,-25.470,-2.099,0.000,0.000,0.000,1.000,1,1,1,PC,,vtkMRMLScalarVolumeNode1
```

Note the two commas following label description (AC, PC) and missing label (1, 2).

NA: To discuss potential implementation of mismatched columns. Currently, this error bypasses current checks. Correct number of columns, but label description was in the place of label, and description is empty.

## Environment

* Browser (with version): Safari (13.1.1) and Chrome (83.0.4103.97)
* Dropdown menu option: Any option
* File uploaded (if applicable): [2009bMNI_T1_TM_01.txt](https://github.com/afids/afids-validator/files/4772594/2009bMNI_T1_TM_01.txt).

_PR template was adopted from [appium](https://github.com/appium/appium/blob/master/.github/ISSUE_TEMPLATE.md)_

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

Resolves #93 

Error is thrown if `row_label` is not numerical / `row_desc` is empty. Adds check for both of these. 
Recommendation is to merge following #101 

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [x] Changes have been tested to ensure that fix is effective or that a feature works.
- [x] Changes pass the unit tests
- [x] Code has been run through [`black`](https://github.com/psf/black) with the `-l 79` flag.
- [x] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

_PR template was adopted from [appium](https://github.com/appium/appium/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_

### CODE REVIEW

**Comment by kaitj** [comment]
Just realized this includes all the commits from the `afids-class` PR so definitely don't merge until #101 is merged!

File: model.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+    expected_label = 1
+    for row in _skip_first(csv_reader, 3):
+        if expected_label > 32:
+            raise InvalidFileError("Too many rows")
+
+        row_label = parse_fcsv_field(row, "label")
+
+        # Check if label is numerical
+        if not row_label.isdigit():
```
Comment (tkkuehn) [inline model.py:131]: I'm a little surprised we don't already just check something like `row_label == str(expected_label)`, which would also cover this case. Assuming there's a good reason for that and I've forgotten it, this makes sense.

File: model.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+            raise InvalidFileError(
+                f"Row label {row_label} is invalid for fiducial {expected_label}"
+            )
+
+        expected_label += 1
+        row_desc = parse_fcsv_field(row, "desc", row_label)
+
+        # Check to see if row description is valid
```
Comment (tkkuehn) [inline model.py:139]: I would maybe adjust this to say "Check to see row description is not empty or None".

File: model.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        if not row_desc:
+            raise InvalidFileError(
+                f"Row description {row_desc} is not a invalid for {row_label}"
```
Comment (tkkuehn) [inline model.py:142]: Typo

**Review by tkkuehn — CHANGES_REQUESTED** [review]
A couple of nitpicks but basically looks good.

If possible, it would be nice to change this PR to merge into the `afids-class` branch instead of `master` -- I think that would get rid of all the irrelevant commits.

**Comment by kaitj** [comment]
> A couple of nitpicks but basically looks good.
> 
> If possible, it would be nice to change this PR to merge into the `afids-class` branch instead of `master` -- I think that would get rid of all the irrelevant commits.

Done!

**Review by tkkuehn — APPROVED** [review]
Looks good to me. Let's just wait until the AFIDs class is merged then merge this to master.

**Pull request merged**